src/app/my-messages/my-messages.component.ts

```diff
@@ -0,0 +1,101 @@
+import { MessagesService } from './messages.service';
+import { DeleteConfirmationDialogComponent } from '../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { MatDialog } from '@angular/material';
+import { AbstractNotification } from './abstract-notification';
+import { CommunityInvitationNotification } from './community-invitation-notification';
+import { JoinCommunityRequestNotification } from './join-community-request-notification';
+import { GlobalErrorHandlerService } from '../error/global-error-handler.service';
+
+@Component({
+  selector: 'app-my-messages',
+  templateUrl: './my-messages.component.html',
+  styleUrls: ['./my-messages.component.scss']
+})
+export class MyMessagesComponent implements OnInit {
+
+  errorMessage: string = null;
+  notifications$ = of([]);
+
+  constructor(private communityRegistrationService: CommunityRegistrationService,
+              private messageService: MessagesService,
+              public dialog: MatDialog,
+              private changeDetector: ChangeDetectorRef,
+              private globalErrorHandlerService: GlobalErrorHandlerService) {
+  }
+
+  ngOnInit() {
+   this.notifications$ = this.messageService.getUserNotifications();
+  }
+
+  hasJoinRequestType<T extends AbstractNotification>(notification: T): boolean {
+    return notification.hasCommunityInvitationType() || notification.hasJoinCommunityType();
+  }
+
+  showAcceptDeclineButtons<T extends AbstractNotification>(notification: T) {
+    if (notification.hasCommunityInvitationType()) {
+      const registration = (<CommunityInvitationNotification><any>notification).registration;
+      return registration.approvedByUser === null || registration.approvedByCommunity === null;
+    } else if (notification.hasJoinCommunityType()) {
+      const registration = (<JoinCommunityRequestNotification><any>notification).registration;
+      return registration.approvedByUser === null || registration.approvedByCommunity === null;
+    } else {
+      return false;
+    }
+  }
+
+  onAccept<T extends AbstractNotification>(notification: T) {
+    const registration = notification.hasCommunityInvitationType() ?
+      (<CommunityInvitationNotification><any>notification).registration :
+      (<JoinCommunityRequestNotification><any>notification).registration;
+    const requestData: CommunityUserRegistration = {
+      id: registration.id,
+      approvedByUser: true,
+      approvedByCommunity: true
+    };
+    this.updateRegistration(registration.community.id, requestData);
+  }
+
+  onDecline<T extends AbstractNotification>(notification: T) {
+    const dialogRef = this.dialog.open(DeleteConfirmationDialogComponent, {
+      width: '350px',
+      data: {
+        message: 'Are you sure you want to decline the request?'
+      }
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        const registration = notification.hasCommunityInvitationType() ?
```

Comment:
Can we use `Util.createNotificationInstance` here instead of explicit casting?

Reply:
thank you. done